fsspec/implementations/tests/test_memory.py

```diff
@@ -159,3 +163,86 @@ def test_remove_all(m):
     m.touch("afile")
     m.rm("/", recursive=True)
     assert not m.ls("/")
+
+
+@pytest.mark.parametrize("funcname", ["cp", "get", "put"])
+def test_cp_get_put_directory_recursive(m, tmpdir, funcname):
```

Comment:
I aimed to keep this test the same as the equivalent one in `test_local.py` but it has become rather complex to get it to work on Windows. I could always split it into 3 constituent tests (`cp`, `get` and `put`).

Reply:
I like splitting it